app/tasks/batch.py

```diff
@@ -1,31 +1,37 @@
-from typing import Any, Dict, List, Optional, Callable, Awaitable, Set
-from time import sleep
+import logging
 from datetime import datetime
-import os
-
-from app.models.pydantic.job import Job
-
-POLL_WAIT_TIME = 30
-BATCH_CLIENT = None
-REGION = os.environ.get("REGION", "us-east-1")
-
-
-def execute(jobs: List[Job], callback: Callable[[Dict[str, Any]], Awaitable[None]]) -> None:
-    scheduled_jobs = schedule(jobs)
-
-    return poll_jobs(scheduled_jobs.values(), callback)
-
-
-def get_batch_client():
-    import boto3
-
-    global BATCH_CLIENT
-    if BATCH_CLIENT is None:
-        BATCH_CLIENT = boto3.client("batch", region_name=REGION)
-    return BATCH_CLIENT
-
-
-def schedule(jobs: List[Job]) -> Dict[str, str]:
+from time import sleep
+from typing import Any, Awaitable, Callable, Dict, List, Optional, Set
+
+from ..models.pydantic.change_log import ChangeLog
+from ..models.pydantic.jobs import Job
+from ..settings.globals import POLL_WAIT_TIME
+from ..utils.aws import get_batch_client
+
+
+async def execute(
+    jobs: List[Job], callback: Callable[[Dict[str, Any]], Awaitable[None]]
```

Comment:
I think I'm the one who put this in here based on what you were doing in a different class, but should we actually use callbacks like this? It's being used specifically to add to the change log, and wouldn't work for any other function really. It should probably be more clear in this is just a change logger.

Reply:
maybe just pass some kind of StatusLogger object, which accepts the basic fields (status/message/detail) and the caller will instantiate object with whatever other details are needed to log correctly (e.g. dataset/version)?

Reply:
We should probably make this its own feature. I will create a JIRA ticket for this and cover this in a different PR